package/com.unity.formats.usd/Runtime/Scripts/IO/Materials/ShaderExporterBase.cs

```diff
@@ -33,33 +33,120 @@ public class ShaderExporterBase {
                                  string textureName,
-                                 string textureOutput) {
-#if UNITY_EDITOR
-      var srcPath = UnityEditor.AssetDatabase.GetAssetPath(material.GetTexture(textureName));
-      srcPath = srcPath.Substring("Assets/".Length);
-      srcPath = Application.dataPath + "/" + srcPath;
-      var fileName = System.IO.Path.GetFileName(srcPath);
-      var filePath = System.IO.Path.Combine(destTexturePath, fileName);
-
-      System.IO.File.Copy(srcPath, filePath, overwrite: true);
-
-      // Make file path baked into USD relative to scene file and use forward slashes.
-      filePath = ImporterBase.MakeRelativePath(scene.FilePath, filePath);
-      filePath = filePath.Replace("\\", "/");
-
-      var uvReader = new PrimvarReaderSample<Vector2>();
-      uvReader.varname.defaultValue = new TfToken("st");
-      scene.Write(usdShaderPath + "/uvReader", uvReader);
-      var tex = new TextureReaderSample(filePath, usdShaderPath + "/uvReader.outputs:result");
-      scene.Write(usdShaderPath + "/" + textureName, tex);
-      return usdShaderPath + "/" + textureName + ".outputs:" + textureOutput;
-#else
-      // Not supported at run-time, too many things can go wrong
-      // (can't encode compressed textures, etc).
-      throw new System.Exception("Not supported at run-time");
-#endif
-    }
+                                 string textureOutput)
+    {
+      // We have to handle multiple cases here:
+      // - file exists on disk
+      //   - file is a supported format => can be directly copied
+      //   - file is not in a supported format => need to blit / export
+      // - file is only in memory
+      //   - a Texture2D
+      //   - a Texture
+      //   - a RenderTexture
+      //   - needs special care if marked as Normal Map
+      //     (can probably only be detected in an Editor context, and heuristically at runtime)
+      //   => need to blit / export
+      // - file is not supported at all (or not yet)
+      //   - a 3D texture
+      //   => needs to be ignored, log Warning
+
+      bool textureIsExported = false;
+
+      string filePath = null;
+      string fileName = null;
+
+      var texture = material.GetTexture(textureName);
+
+  #if UNITY_EDITOR
+      var srcPath = UnityEditor.AssetDatabase.GetAssetPath(texture);
+      if(!string.IsNullOrEmpty(srcPath))
+      { 
+        srcPath = srcPath.Substring("Assets/".Length);
+        srcPath = Application.dataPath + "/" + srcPath;
+        fileName = System.IO.Path.GetFileName(srcPath);
+        filePath = System.IO.Path.Combine(destTexturePath, fileName);
+
+        if(System.IO.File.Exists(srcPath))
+        { 
+          // USD officially only supports png / jpg / jpeg
+          // https://graphics.pixar.com/usd/docs/Usdz-File-Format-Specification.html
 
+          var ext = System.IO.Path.GetExtension(srcPath).ToLowerInvariant();
+          if(ext == ".png" || ext == ".jpg" || ext == ".jpeg") { 
```

Comment:
I can't find the code now, but I feel like USD supports the esoteric jpeg extensions, like jfif, etc.

Reply:
True, USD also supports TGA/TIF/PSD etc.
But USDZ only supports PNG/JPG (https://graphics.pixar.com/usd/docs/Usdz-File-Format-Specification.html), so best way would be to distinguish here in which kind of export we currently are (ExportContext?) - what would be the best way to approach that?

Reply:
Oh, I just meant that "jfif" IS actually a jpeg file, it's just an unusual extension that most people don't know about.

I think it's actually fine to only export png/jpg because I've seen issues with USD loading files like PDFs anyway -- sometimes it works, sometimes not (maybe bugs in OpenImageIO).

Anyway, I think this is a good default without adding more options to ExportContext, I'm just suggesting to add all the supported JPEG file extensions.

Reply:
I see. In above spec link it's mentioned: "png and jpeg files (any of the multiple common extensions for jpeg)".
I still think this needs to be configurable (maybe not in this PR) since otherwise some roundtrip stuff breaks (moving usd out of Unity and back in will have different textures), not sure how important that is.